### PR TITLE
Host NFD Helm repo in gh-pages

### PR DESCRIPTION
**Pull request by marquiz**
- scripts/update-gh-pages: manage a Helm repo
  - Make the update-gh-pages.sh script to maintain a Helm charts repository
under charts/ subdirectory in gh-pages. The script now (always) scans
throught all release assets and injects any found Helm chart archives
into the Helm repo. In practice, new assets in all Github releases are
scanned and the Helm repo is updated on any update of the master or
release branches or on any new tags. Asset ids are tracked/cached in
order to avoid unnecessary downloads, but also, to capture any changes
in assets that were already merged in the repo index.
  - After this a user is able to do
    ```
    $ helm repo add nfd http://kubernetes-sigs.github.io/node-feature-discovery/charts
      ...
    $ helm repo update
      ...
    $ helm install nfd/node-feature-discovery --namespace nfd --create-namespace --generate-name
      ...
    ```
- scripts/prepare-release: package Helm chart
  Prepare a signed Helm chart package to be uploaded to the Github release page.
- update release process in new-release issue template
   Capture process for Helm charts and remove some outdated bits.

I ended up this solution because:
- `gh-pages` branch stays fully automated. No need for manual pushes/PRs
- Release artefacts (Helm chart archives) can be updated after the fact. If I would've automated the creation of Helm chart archives (based on Git tags) in gh-pages this would not have been possible.

### CODE REVIEW

**Comment by marquiz** [comment]
Needs some more testing...
/hold

**Comment by marquiz** [comment]
Works in my personal repo:
https://github.com/marquiz/node-feature-discovery/runs/1988617946?check_suite_focus=true

This deployment using my personal Helm repo works:
```
$ helm repo add nfd https://marquiz.github.io/node-feature-discovery/charts

$ helm repo update

$ helm install nfd/node-feature-discovery --namespace nfd --create-namespace --generate-name
```

I think this is ready for review. It'd be good if someone more acquainted with Helm than me (e.g. @adrianchiris) would take a look 😉 
/unhold

**Comment by adrianchiris** [comment]
Awesome @marquiz ! i will definitely dig into this PR.

File: scripts/prepare-release.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@ this=`basename $0`
 
 usage () {
 cat << EOF
-Usage: $this [-h] RELEASE_VERSION
+Usage: $this [-h] RELEASE_VERSION GPG_KEY GPG_KEYRING
```
Comment (adrianchiris) [inline scripts/prepare-release.sh:8]: So OWNERS (who wish to do a release) are required to generate a GPG key first ?
should this GPG key be associated with the github account ?

File: scripts/github/update-gh-pages.sh [hunk 2]
```diff
@@ -22,11 +22,59 @@ create_versions_js() {
...
+            # Set fields we're interested in as "asset_<field>" shell variables
+            eval `echo $asset_meta | jq -r '{id, name, url, browser_download_url} | keys[] as $k | "local asset_\($k)=\(.[$k])"'`
+
+            if [[ "$asset_name" != node-feature-discovery-chart-*tgz ]]; then
+                echo "Asset $asset_name does not look like a Helm chart archive, skipping..."
+                continue
+            fi
+
+            # Check if the asset has changed
+            asset_id_old=`cat "$asset_name".id 2> /dev/null || :`
```
Comment (adrianchiris) [inline scripts/github/update-gh-pages.sh:54]: do you expect changes to existing assets ?

im wondering if its possible to just query draft releases and update index.yaml for those only.

**Review by adrianchiris — COMMENTED** [review]
Hi there @marquiz ! Sorry for the late response.

Overall, i think this change is good.
A couple of (naive) comments. 

my next step is to fork and play around with this change.

File: scripts/github/update-gh-pages.sh (hunk 2 above)
Comment (marquiz) [reply scripts/github/update-gh-pages.sh:54]: Uploading of the assets is a manual process so I would keep this as a way correct mistakes, e.g. the maintainer fatfingering something

File: scripts/prepare-release.sh (hunk 1 above)
Comment (marquiz) [reply scripts/prepare-release.sh:8]: Yes, it should. I don't know if there's a way to verify this 🤔

Comment (marquiz) [reply scripts/prepare-release.sh:8]: Thanks for trying this out @adrianchiris!

Yes, I used the key uid reported by `gpg --list-secret-keys`

I.e. something like
```
./scripts/prepare-release.sh v0.8.0 'Markus Lehtonen <markus.lehtonen@intel.com>' ~/.gnupg/secring.gpg
```

Comment (marquiz) [reply scripts/prepare-release.sh:8]: @adrianchiris: added an example to `prepare-release.sh` usage string

Comment (adrianchiris) [reply scripts/prepare-release.sh:8]: Mmm... did exactly the same but it complains the key was not found even though i see it in my secret keyring.

maybe i have some env issue.

Comment (marquiz) [reply scripts/prepare-release.sh:8]: Hmm, strange 🤔 Don't put the `[ultimate]` there, if you have it

Comment (marquiz) [reply scripts/prepare-release.sh:8]: Ach, actually the `secring.gpg` file is obsolete and used by older versions of gpg. You prolly don't have your key there

Comment (marquiz) [reply scripts/prepare-release.sh:8]: Update the script even more notes

**Comment by marquiz** [comment]
@adrianchiris have you had the time to look at this?

**Comment by adrianchiris** [comment]
not since our last slack discussion, will get to it this week.

**Comment by marquiz** [comment]
> not since our last slack discussion, will get to it this week.

That would be great as this is basically the last open PR blocking the release

**Review by adrianchiris — COMMENTED** [review]
OK, not a bash expert here but the code seems OK to me.

i have also tested the changes by doing a dummy release on my forked repo ensuring the chart can be downloaded and is of the correct version.

I wasnt able to verify the package with provenance file unfortunately, but that may be related to my gpg setup and helm quirks.

@marquiz  were you able do do a helm verify on the package ? 

other than that im lgtm on this.

**Comment by marquiz** [comment]
> I wasnt able to verify the package with provenance file unfortunately, but that may be related to my gpg setup and helm quirks.
> 
> @marquiz were you able do do a helm verify on the package ?

Thanks for the review @adrianchiris! I thought I correctly verified the `--verify` option but must have done that locally with incorrect archive or fatfingered in some other way because now it failed 😬  The root cause for this is the filename mangling we do in the release script. I pushed a new patch constructing the provenance file by and and using `gpg` for signing it. Improves user experience 😄 
 Haven't yet verified that, though. Will report back when I have

**Review by adrianchiris — COMMENTED** [review]
pulled the change,

after creating the chart package:

```
helm verify ./node-feature-discovery-chart-0.8.0.tgz
Signed by: Adrian Chiris <adrianc@nvidia.com>
Using Key With Fingerprint: 7E3B1541735B7D751CD2BB1BCC911D9D88FC463E
Chart Hash Verified: sha256:08932a483344cb5e3ef501083253fc8ab7bfe00f0463e6c06a2fe1bfa2851abd
```

**Comment by marquiz** [comment]
Yeah, I also verified this locally. Not with `install --verify` yet

**Comment by marquiz** [comment]
@adrianchiris verified install from my personal repo now works:
```
$ helm install --generate-name --namespace nfd-helm --create-namespace --verify nfd/node-feature-discovery
NAME: node-feature-discovery-1615973606
LAST DEPLOYED: Wed Mar 17 11:33:29 2021
NAMESPACE: nfd-helm
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

**Comment by adrianchiris** [comment]
Awsome !

/lgtm

**Comment by kfox1111** [comment]
Awesome. :)

Can someone add a reference to the repo to artifacthub.io please?

**Pull request merged**